File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        maximum = 0
        for num in nums:
            if num - 1 not in nums_set:
                i = 1
                while True:
                    if num + i not in nums_set:
                        break
                    i += 1
                maximum = max(maximum, i)
        
        return maximum
        

File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = INT_MIN, currSum = INT_MIN;

        for (const auto n : nums) {
            if (currSum < 0) {
                currSum = n;
            } else {
                currSum += n;
            }

            maxSum = std::max(maxSum, currSum);
        }

        return maxSum;
    }
};
